### ISSUE TITLE
Segment is not sending data to Firebase IOS

### ISSUE THREAD

**Issue by alaminopu** [expo/expo#10088]
### Summary of Issue 
In our Expo managed app, we are using `expo-analytics-segment` to send data to segment and from there to other destination like Mixpanel, Firebase etc. 

It works pretty well except Segment doesn't send data to Firebase IOS destination. However, it does send all data to Firebase Android. 

I am assuming the underlying [integration](https://segment.com/docs/connections/destinations/catalog/firebase/#getting-started-on-ios) with the SDK is correct and maybe a bug there? 

### Environment - output of `expo diagnostics` & the platform(s) you're targeting
```
  Expo CLI 3.24.0 environment info:
    System:
      OS: macOS Mojave 10.14.5
      Shell: 5.3 - /bin/zsh
    Binaries:
      Node: 14.9.0 - /usr/local/bin/node
      Yarn: 1.22.5 - /usr/local/bin/yarn
      npm: 6.14.8 - /usr/local/bin/npm
      Watchman: 4.9.0 - /usr/local/bin/watchman
    IDEs:
      Android Studio: 3.5 AI-191.8026.42.35.5791312
      Xcode: 11.3.1/11C504 - /usr/bin/xcodebuild
    npmPackages:
      expo: ^38.0.9 => 38.0.9
      react: 16.11.0 => 16.11.0
      react-native: https://github.com/expo/react-native/archive/sdk-38.0.2.tar.gz => 0.62.2
      react-navigation: ^4.0.10 => 4.4.0
    npmGlobalPackages:
      expo-cli: 3.24.0
```

**Comment by cruzach** [comment]
Looks like this was added on Android - https://github.com/expo/expo/pull/3615 - but not yet on iOS

We can keep this open as an enhancement for `expo-analytics-segment`

**Comment by Faradey27** [comment]
Hey @cruzach , 
any news on the IOS part of segment-firebase integration?

**Comment by brentvatne** [comment]
@Faradey27 - we aren't working on that, working on making it possible to include any arbitrary native code instead: https://blog.expo.io/expo-managed-workflow-in-2021-d1c9b68aa10

**Comment by cruzach** [comment]
FYI Segment has recently released a brand new react native library (mentioned above) that is Expo-compatible, and it is likely that we will deprecate `expo-analytics-segment` in favor of their new library. 

I recommend trying out this new library if you're looking for the best Segment experience: https://github.com/segmentio/analytics-react-native/blob/master/README.md

**Comment by MorenoMdz** [comment]
> FYI Segment has recently released a brand new react native library (mentioned above) that is Expo-compatible, and it is likely that we will deprecate `expo-analytics-segment` in favor of their new library.
> 
> I recommend trying out this new library if you're looking for the best Segment experience: https://github.com/segmentio/analytics-react-native/blob/master/README.md

Quick question about the EAS and custom dev clients, we haven't started using those yet, and from the first readings about it I am a bit confused. If we decide to use the new Segment RN library will it limit our ability to use the Expo Go client? We use it a lot for development but as well for QA.

**Comment by cruzach** [comment]
you should be able to use the Expo Go client, there's only one native method in the library and it's not required, it just gets device info which is all doable from `expo-device` anyways

**Comment by MorenoMdz** [comment]
> you should be able to use the Expo Go client, there's only one native method in the library and it's not required, it just gets device info which is all doable from `expo-device` anyways

Thanks, in this case I would need to add the device programmatically into the Segment payload instead of using expo-device, is that correct?

**Comment by axelnormand** [comment]
I'm not sure one can use segment v2 with Expo Go (non-EAS)?

I get an error saying "Failed to load Analytics native module". Looking at the segment v2 source, all the identify, track etc methods are called over the bridge. Am i missing something?

**Comment by cruzach** [comment]
the native module has very few methods, it's mostly a JS library: https://github.com/segmentio/analytics-react-native/blob/master/packages/core/android/src/main/java/com/segmentanalyticsreactnative/AnalyticsReactNativeModule.kt

**Comment by axelnormand** [comment]
I see, yes. Great thanks for info. Perhaps i had some cached JS lying around from the old version that was calling over the bridge. Let me try again and report back
